src/main/java/com/appdynamics/monitors/azure/metrics/AzureMetrics.java

```diff
@@ -0,0 +1,185 @@
+                                break;
+                            case "Maximum":
+                                metricValue = String.valueOf(metricCollection.metrics().get(0).timeseries().get(0).data().get(0).maximum());
+                                break;
+                            default:
+                                logger.info("Not Reporting Metric {} for Resource {} as the aggregation type is not supported",
+                                        metricName,
+                                        resource.name());
+                                break;
+                        }
+                        assert metricValue != null;
+                        if (metricValue.isEmpty() || metricValue.equals("0.0") || metricValue.equals("null")) {
+                            logger.debug("Ignoring Metric {} for Resource {} as it is null or empty",
+                                    metricName,
+                                    resource.name(),
+                                    metricValue);
+                        } else {
+                            Metric metric = new Metric(metricName, metricValue, metricPath + metricName);
+                            if (logger.isDebugEnabled()) {
+                                logger.debug("Reporting Metric {} for Resource {} with value {}",
+                                        metricName,
+                                        resource.name(),
+                                        metricValue);
+                            }
+                            finalMetricList.add(metric);
+                        }
+                    } else {
+                        if (logger.isDebugEnabled()) {
+                            logger.debug("Not Reporting Metric {} for Resource {} as it is filtered by {}",
+                                    resourceMetric.name().value(),
+                                    resource.name(),
+                                    currentMetricFilter);
+                        }
+                    }
+                }
+            }
+        } else {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Skipping Resource {} of Type {} in Group {} because of Resource Filter {} of Type Filter {} in Group Filter {}",
+                        resource.name(),
+                        resource.resourceType(),
+                        resource.resourceGroupName(),
+                        currentResourceTypeFilter,
+                        currentResourceTypeFilter,
+                        currentResourceGroupFilter);
+            }
+        }
+        if (finalMetricList.isEmpty()) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Metric List is empty");
+            }
+        } else {
+            metricWriteHelper.transformAndPrintMetrics(finalMetricList);
```

Comment:
We have started publishing a 'Heart Beat' metric per configured instance. It's a 0/1 metric that prints 0 in case of an error/connectivity issues etc and 1 in case of successful metric collection. You can refer to the Netscaler Extension for how this is done.